whisper/stt/processing/vad.py

```diff
@@ -0,0 +1,390 @@
+import numpy as np
+import os
+import shutil
+from stt import logger, USE_CTRANSLATE2
+
+
+_silero_vad_model = {}
+_has_onnx = None
+_vad_import = None
+
+
+def remove_non_speech(
+    audio,
+    use_sample=False,
+    min_speech_duration=0.1,
+    min_silence_duration=1,
+    dilatation=0.5,
+    sample_rate=16000,
+    method="silero",
+    avoid_empty_speech=False,
+    return_format="tuple",
+):
+    """
+    Remove non-speech segments from audio (using Silero VAD),
+    glue the speech segments together and return the result along with
+    a function to convert timestamps from the new audio to the original audio
+
+    parameters:
+        audio: torch.Tensor
+            audio data *in 16kHz*
+        use_sample: bool
+            if True, return start and end in samples instead of seconds
+        min_speech_duration: float
+            minimum duration (in sec) of a speech segment
+        min_silence_duration: float
+            minimum duration (in sec) of a silence segment
+        dilatation: float
+            how much (in sec) to enlarge each speech segment detected by the VAD
+        method: str
+            method to use to remove non-speech segments
+        avoid_empty_speech: bool
+            if True, avoid returning an empty speech segment (re)
+    """
+
+    if USE_CTRANSLATE2 and method == "silero":
+        from faster_whisper.vad import VadOptions
+
+        options = VadOptions(
+            min_speech_duration_ms=min_speech_duration * 1000,
+            min_silence_duration_ms=min_silence_duration * 1000,
+        )
+        from faster_whisper.vad import get_speech_timestamps
+
+        segments = get_speech_timestamps(audio, vad_options=options)
+    else:
+        segments = get_vad_segments(
+            audio,
+            sample_rate=sample_rate,
+            output_sample=True,
+            min_speech_duration=min_speech_duration,
+            min_silence_duration=min_silence_duration,
+            dilatation=dilatation,
+            method=method,
+        )
+    segments = [(seg["start"], seg["end"]) for seg in segments]
+    if len(segments) == 0:
+        if avoid_empty_speech:
+            segments = [(0, audio.shape[-1])]
+        else:
+            np.array([]), [], lambda t, t2=None: t if t2 is None else [t, t2]
+
+    audio_speech = np.concatenate([audio[..., s:e] for s, e in segments], axis=-1)
+    # audio_speech = torch.cat([audio[..., s:e] for s,e in segments], dim=-1)
```

Comment:
remove the commented out line.

And move this below the "`if return_format == "dict"`".
It's better to avoid computing this when possible (to save RAM and time)